### PR TITLE
ACI-CNI Pods - Provide granular priorityclass options

### PR DESCRIPTION
**Pull request by pkharat**

### CODE REVIEW

File: provision/testdata/with_overrides.kube.yaml [hunk 1]
```diff
@@ -2024,6 +2030,7 @@ spec:
       tolerations:
         - operator: Exists
       priorityClassName: system-cluster-critical
+      priorityClassName: aci-containers-host
```
[the remark this answers is not in the document]
Comment (pkharat) [reply provision/testdata/with_overrides.kube.yaml:2033]: The generated yaml for with_overrides test is not correct, it is setting both the priority class names. The flavor used in this test is kubernetes-1.25. Checking how to fix this.

**Comment by snaiksat** [comment]
> How would this flow work ? When user want to use these priority classes, user has to set the flag as "use_acicni_priority_class: False" in flavors.yaml 

No, the user never touches flavors.yaml (that file is part of the source code). The suggested logic in our template should take care of overriding that setting, no?

  {% if not config.kube_config.no_priority_class %}
  priorityClassName: system-node-critical
  {% endif %}
  {% if config.kube_config.use_acicni_priority_class %}
  priorityClassName: acicni-priority
  {% if config.kube_config.use_aci_containers_controller_priority_class %}
  priorityClassName: acicni-containers-controller
  {% endif %}

File: provision/acc_provision/templates/aci-containers.yaml [hunk 2]
```diff
@@ -2996,3 +3005,57 @@ spec:
...
+---
+{% set flavor_version = config.flavor.split('-') %}
+{% set major_version = flavor_version[1].split('.')[0] | int %}
+{% set minor_version = flavor_version[1].split('.')[1] | int %}
+{% if (major_version >= 4 and  minor_version >= 9
+       and flavor_version[2] in ["openstack", "esx", "baremetal"]) %}
+apiVersion: scheduling.k8s.io/v1
+{% else %}
+apiVersion: scheduling.k8s.io/v1beta1
+{% endif %}
```
Comment (snaiksat) [inline provision/acc_provision/templates/aci-containers.yaml:3018]: Can we just do this processing once in the config build stage, set the api version string as a config, and use it here?

**Review by snaiksat — CHANGES_REQUESTED** [review]

**Comment by pkharat** [comment]
> > How would this flow work ? When user want to use these priority classes, user has to set the flag as "use_acicni_priority_class: False" in flavors.yaml
> 
> No, the user never touches flavors.yaml (that file is part of the source code). The suggested logic in our template should take care of overriding that setting, no?
> 
> ```
> {% if not config.kube_config.no_priority_class %}
> priorityClassName: system-node-critical
> {% endif %}
> {% if config.kube_config.use_acicni_priority_class %}
> priorityClassName: acicni-priority
> {% if config.kube_config.use_aci_containers_controller_priority_class %}
> priorityClassName: acicni-containers-controller
> {% endif %} 
> ```

Overriding is not happening. As I pointed out, always 2 priority class fields are set.
In case of openshift this one (bec flavour.yaml sets "use_acicni_priority_class: True" and "no_priority_class: True")
```
      priorityClassName: acicni-priority
      priorityClassName: aci-containers-host
```

and in case of k8s (1.25) (it has "use_acicni_priority_class: False" and "no_priority_class: False")
```
      priorityClassName: system-cluster-critical
      priorityClassName: aci-containers-host
```

It's because of the if conditions in the containers.yaml. Either one of it is executed for openshift/k8s
```
{% if not config.kube_config.no_priority_class %}
priorityClassName: system-node-critical
{% endif %}
{% if config.kube_config.use_acicni_priority_class %}
priorityClassName: acicni-priority
{% endif %}
```  
And after this the priority class if condition is also executed ending up setting 2 priority class field names.

On further debugging, observed that this issue is seen in kube.yaml only.

Other resource specific yaml in tar folder e.g cluster-network-31-DaemonSet-aci-containers-host.yaml, cluster-network-32-DaemonSet-aci-containers-openvswitch.yaml, cluster-network-33-Deployment-aci-containers-controller.yaml
it has only one priority class field set as expected.

**Comment by pkharat** [comment]
This branch has conflicts with the master. Did pull master branch changes into this branch and ran fix-testdata.sh, but tests are failing. So(because of time constraints) better closing this PR and creating new branch from master and porting changes to newer branch.

**Comment by pkharat** [comment]
New PR - https://github.com/noironetworks/acc-provision/pull/961
Addressed all the review comments and changes added to the new PR.

**Pull request closed without merging**